### PR TITLE
NO-ISSUE: Specify 'strategy: Recreate' to all ReviewApp's manifests

### PR DESCRIPTION
**Pull request by ShotaKitazawa**
# Description

No Issue

Specify `strategy: Recreate` to ensure that the ReviewApps' target Pods are updated.

## Type of change

- [ ] Document update or simple typo fix
- [ ] Maintenance/update components
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [ ] I have checked backward/forward compatibility that may cause regarding this change.

### CODE REVIEW

File: manifests/app/dreamkast/overlays/development/template-dk/.patch_deployment-strategy.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: dummy
+spec:
+  # ReviewApp 対象の Pod が必ず更新されるよう strategy に Recreate を指定
+  strategy:
+    rollingUpdate:
```
Comment (jacopen) [inline manifests/app/dreamkast/overlays/development/template-dk/.patch_deployment-strategy.yaml:8]: type: RecreateするのであればこのrollingUpdateはいらないかも？

Comment (ShotaKitazawa) [reply manifests/app/dreamkast/overlays/development/template-dk/.patch_deployment-strategy.yaml:8]: 既存の `spec.strategy.rollingUpdate` の記述を kustomize の patches で消すために、意図的に書いています

Comment (ShotaKitazawa) [reply manifests/app/dreamkast/overlays/development/template-dk/.patch_deployment-strategy.yaml:8]: ちなみに type: Recreate で rollingUpdate を指定すると怒られます (偉い)

```
$ k apply -f tmp.yaml
The Deployment "nginx-kanata" is invalid: spec.strategy.rollingUpdate: Forbidden: may not be specified when strategy `type` is 'Recreate'
```

**Comment by ShotaKitazawa** [comment]
確認用のコマンドです

```bash
ARGOCD_ENV_NAMESPACE=hoge ARGOCD_ENV_FQDN=example.com ARGOCD_ENV_BASE_URL=https://example.com kustomize build manifests/app/dreamkast/overlays/development/template-ui/ 
```

**Review by jacopen — APPROVED** [review]

**Pull request merged**